practice/earthquake/site.js

```diff
@@ -0,0 +1,186 @@
+'use strict';
+
+var MS_IN_WK = 1000 * 60 * 60 * 24 * 7;
+var DOT_SCALE = 50000;  /// Good number is 50000
+var DOT_COLOR = '200, 128, 128';  ///'R, G, B'
+
+/**
+ * Makes Atom linter happier about the ol functions.
+ */
+if (!window.ol) {
+  var ol;
+}
+
+/**
+ * Function to load the earthquake data.  Uses a Promise.
+ * @return {[json]} The data in JSON format
+ */
+function getEarthquakeJSON() {
+  var url = 'http://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson';
+  return Promise.resolve($.ajax({
+    dataType: 'json',
+    url: url,
+  }));
+}
+
+
+/**
+ * Create an OpenLayers map and return its identifier.
+ */
+function createMap() {
+  var mapBackground = new ol.layer.Tile({
+    source: new ol.source.OSM()
+  });
+  var map = new ol.Map({
+    layers: [mapBackground],
+    target: 'map',
+    controls: ol.control.defaults({
+      attributionOptions: {
+        collapsible: false
+      }
+    }),
+    view: new ol.View({
+      center: [0, 0],
+      zoom: 2
+    })
+  });
+  $('#map').data('map', map);
+  return map;
+}
+
+/**
+ * Create an empty source to add the earthquate icons to.
+ * @param  {ol.Map}      The Map element that will hold the source.
+ * @return {ol.Source}   The source that will hold the icons.
+ */
+function createSource(map) {
+  var vectorSource = new ol.source.Vector();
+  var earthquakeLayer = new ol.layer.Vector({source: vectorSource});
+  map.addLayer(earthquakeLayer);
+  return vectorSource;
+}
+
+/**
+ * Function to parse the JSON file and gather location, mag, and time/Date
+ *   Filters out data with null values.
+ * @param  {[JSON]}   The data from the earthquake reporting service
+ * @return {[array]}  Array of arrays: [long, lat, mag, UTC]
+ */
+function scrapeEarthquakeData(json) {
+  var data = _.map(json.features, function(feature) {
+    return [feature.geometry.coordinates[0],
+            feature.geometry.coordinates[1],
+            feature.properties.mag,
+            feature.properties.time
+          ];
+  });
```

Comment:
I think this data transformation makes the resulting code harder to read. It's taking nicely named properties and then removing their names and putting them in an array indexed with magic numbers. Even if most of the properties on `features` you don't care about in rendering, the rendering code will be much easier to read with those names.